local/include/core/ProxyVerifier.h

```diff
@@ -80,6 +82,35 @@ static const std::string YAML_RULE_CONTAINS{"contains"};
 static const std::string YAML_RULE_PREFIX{"prefix"};
 static const std::string YAML_RULE_SUFFIX{"suffix"};
 
+enum class YamlUrlPart { Scheme, Host, Port, Authority, Path, Query, Fragment, Error };
+
+static const swoc::Lexicon<YamlUrlPart> NAME_TO_URL_PART{
+    {{YamlUrlPart::Scheme, {"scheme"}},
+     {YamlUrlPart::Host, {"host"}},
+     {YamlUrlPart::Port, {"port"}},
+     {YamlUrlPart::Authority, {"authority", "net-loc"}},
+     {YamlUrlPart::Path, {"path"}},
+     {YamlUrlPart::Query, {"query"}},
+     {YamlUrlPart::Fragment, {"fragment"}}},
+    {YamlUrlPart::Error}};
+
+static const std::string YAML_URL_SCHEME{"scheme"};
+static const std::string YAML_URL_HOST{"host"};
+static const std::string YAML_URL_PORT{"port"};
+static const std::string YAML_URL_AUTHORITY{"net-loc"};
+static const std::string YAML_URL_PATH{"path"};
+static const std::string YAML_URL_QUERY{"query"};
+static const std::string YAML_URL_FRAGMENT{"fragment"};
+
+static const swoc::TextView URL_PART_TO_NAME[] = {
+    swoc::TextView(YAML_URL_SCHEME),
+    swoc::TextView(YAML_URL_HOST),
+    swoc::TextView(YAML_URL_PORT),
+    swoc::TextView(YAML_URL_AUTHORITY),
+    swoc::TextView(YAML_URL_PATH),
+    swoc::TextView(YAML_URL_QUERY),
+    swoc::TextView(YAML_URL_FRAGMENT)};
+
```

Comment:
If I understand things correctly, with swoc::Lexicon, we should no longer need this second map. Lexicon is made for indexing in both directions (indexing from the std::string_view and from the enum).